kafka/client/client.go

```diff
@@ -0,0 +1,89 @@
+package kafkasender
+
+import (
+	"context"
+	"log"
+	"os"
+
+	"github.com/Shopify/sarama"
+	"github.com/cloudevents/sdk-go/protocol/kafka_sarama/v2"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/google/uuid"
+)
+
+//CloudEventsClient is the method signature for Kafka message
+type CloudEventsClient interface {
+	KafkaMessage(event string, userID string, email string, role []string)
+}
+
+//Kafka struct containing the kafka topic and broker
+type Kafka struct {
+	topic  string
+	broker string
+}
+
+//NewKafka creates a kafka struct containing the kafka topic and broker
+func (k *Kafka) NewKafka() *Kafka {
```

Comment:
I recommend splitting this into two functions:  1) one that reads variables from the environment (using envconfig) and produces a struct that just contains the configuration 2) one that creates a Kafka object from the configuration struct.

Reply:
done